File: app/src/main/java/com/sumant/todonotesapp/MyNotesActivity.java
package com.sumant.todonotesapp;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.sumant.todonotesapp.adapter.NotesAdapter;
import com.sumant.todonotesapp.clickListener.ItemClickListener;
import com.sumant.todonotesapp.model.Notes;

import java.util.ArrayList;
import java.util.List;

public class MyNotesActivity extends AppCompatActivity {

    TextView fullNameTv;
    FloatingActionButton floatingActionButton;
    SharedPreferences sharedPreferences;
    String fullName;

    RecyclerView notesRv;
    ArrayList<Notes> notesList = new ArrayList<>() ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my_notes);

        bindViews();
        setupSharedPref();
        getIntentData();

        fullNameTv.setText( fullName );

        floatingActionButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                setupDialogBox();
            }
        });
    }

    private void getIntentData() {
        fullName = getIntent().getStringExtra(AppConstrain.Full_Name);
        if ( TextUtils.isEmpty(fullName) ){
            fullName = sharedPreferences.getString(PrefConstant.FULL_NAME,"");
        }
    }

    private void bindViews() {
        fullNameTv = findViewById(R.id.fullNameTV);
        floatingActionButton = findViewById(R.id.fab);
        notesRv = findViewById(R.id.notesRV);
    }

    private void setupSharedPref() {
        sharedPreferences = getSharedPreferences(PrefConstant.SHARED_PREFERENCES_NAME,MODE_PRIVATE);

    }

    private void setupDialogBox() {
        View view  = LayoutInflater.from(MyNotesActivity.this).inflate(R.layout.add_notes,null);

        final EditText titleEditText = view.findViewById(R.id.titleAddNoteEditText);
        final EditText descriptionEditText = view.findViewById(R.id.descriptionAddNoteEditText);
        Button submitButton = view.findViewById(R.id.submitButton);

        final AlertDialog alertDialog = new AlertDialog.Builder(this)
                .setView(view)
                .create();
        alertDialog.show();

        submitButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String title = titleEditText.getText().toString();
                String description = descriptionEditText.getText().toString();

                if ( !TextUtils.isEmpty(title) && !TextUtils.isEmpty(description) ){
                    Notes notes = new Notes(title,description);
                    notesList.add( notes );
                    setupRecyclerView();
                }else
                    Toast.makeText(MyNotesActivity.this, "Title and Description can't be empty", Toast.LENGTH_SHORT).show();
                alertDialog.hide();
            }
        });
    }

    private void setupRecyclerView() {

        ItemClickListener itemClickListener = new ItemClickListener() {
            @Override
            public void onClick(Notes notes) {
                Intent intent = new Intent(MyNotesActivity.this,DetailActivity.class);
                intent.putExtra(AppConstrain.TITLE,notes.getTitle());
                intent.putExtra(AppConstrain.DESCRIPTION,notes.getDescription());
                startActivity(intent);
            }
        };

        NotesAdapter notesAdapter = new NotesAdapter(notesList,itemClickListener);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(RecyclerView.VERTICAL);
        notesRv.setLayoutManager(linearLayoutManager);
        notesRv.setAdapter(notesAdapter);
    }
}

File: app/src/main/java/com/sumant/todonotesapp/MyNotesActivity.kt
package com.sumant.todonotesapp

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.sumant.todonotesapp.adapter.NotesAdapter
import com.sumant.todonotesapp.clickListener.ItemClickListener
import com.sumant.todonotesapp.model.Notes
import java.util.*

class MyNotesActivity : AppCompatActivity() {
    private lateinit var fullNameTv: TextView
    private lateinit var floatingActionButton: FloatingActionButton
    var sharedPreferences: SharedPreferences? = null
    var fullName: String? = null
    private lateinit var notesRv: RecyclerView
    var notesList = ArrayList<Notes>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_my_notes)
        bindViews()
        setupSharedPref()
        intentData
        fullNameTv.text = fullName
        floatingActionButton.setOnClickListener { setupDialogBox() }
    }

    private val intentData: Unit
        private get() {
            fullName = intent.getStringExtra(AppConstrain.Full_Name)
            if (TextUtils.isEmpty(fullName)) {
                fullName = sharedPreferences!!.getString(PrefConstant.FULL_NAME, "")
            }
        }

    private fun bindViews() {
        fullNameTv = findViewById(R.id.fullNameTV)
        floatingActionButton = findViewById(R.id.fab)
        notesRv = findViewById(R.id.notesRV)
    }

    private fun setupSharedPref() {
        sharedPreferences = getSharedPreferences(PrefConstant.SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE)
    }

    private fun setupDialogBox() {
        val view = LayoutInflater.from(this@MyNotesActivity).inflate(R.layout.add_notes, null)
        val titleEditText = view.findViewById<EditText>(R.id.titleAddNoteEditText)
        val descriptionEditText = view.findViewById<EditText>(R.id.descriptionAddNoteEditText)
        val submitButton = view.findViewById<Button>(R.id.submitButton)
        val alertDialog = AlertDialog.Builder(this)
                .setView(view)
                .create()
        alertDialog.show()
        submitButton.setOnClickListener {
            val title = titleEditText.text.toString()
            val description = descriptionEditText.text.toString()
            if (!TextUtils.isEmpty(title) && !TextUtils.isEmpty(description)) {
                val notes = Notes(title, description)
                notesList.add(notes)
                setupRecyclerView()
            } else Toast.makeText(this@MyNotesActivity, "Title and Description can't be empty", Toast.LENGTH_SHORT).show()
            alertDialog.hide()
        }
    }

    private fun setupRecyclerView() {
        val itemClickListener: ItemClickListener = object : ItemClickListener {
            override fun onClick(notes: Notes?) {
                val intent = Intent(this@MyNotesActivity, DetailActivity::class.java)
                intent.putExtra(AppConstrain.TITLE, notes!!.title)
                intent.putExtra(AppConstrain.DESCRIPTION, notes.description)
                startActivity(intent)
            }
        }
        val notesAdapter = NotesAdapter(notesList, itemClickListener)
        val linearLayoutManager = LinearLayoutManager(this)
        linearLayoutManager.orientation = RecyclerView.VERTICAL
        notesRv!!.layoutManager = linearLayoutManager
        notesRv!!.adapter = notesAdapter
    }
}